src/main/java/com/lasterbergamot/balldontlie/domain/team/service/impl/TeamServiceImpl.java

```diff
@@ -34,14 +38,45 @@ public void getAllTeamsFromBalldontlieAPI() {
     }
 
     @Override
-    public List<Team> getTeams(final int count) {
+    public List<Team> getTeams(final int count, final Conference conference, final Division division) {
         List<Team> teams = teamRepository.findAll();
+        Predicate<Team> teamPredicate = getTeamPredicateAccordingToConferenceAndDivision(conference, division);
+        
+        return teamPredicate == null
+                ? handleCount(count, teams)
+                : handleCount(count, handlePredicate(teams, teamPredicate));
+    }
+
+    private Predicate<Team> getTeamPredicateAccordingToConferenceAndDivision(final Conference conference, final Division division) throws TeamQueryException {
+        Predicate<Team> teamPredicate;
```

Comment:
This method can be reduced to this:
```
if (!division != null && conference != null && division.getConference().equals(conference)) {
  throw new TeamQueryException("The given division is not part of the given conference!");
}
Predicate<Team> divisionPredicate = division == null ? team -> true || team -> team.getDivision().equals(division);
Predicate<Team> conferencePredicate = conference == null ? team -> true || team -> team.getConference().equals(conference);
return divisionPredicate.and(conferencePredicate);
```
This way you don't have to repeat expressions like `team.getDivision().equals(division)` and `null` inputs are handled in a more expressive way. Also we will never return null, so `getTeams` (and other places) can be simplified